### ISSUE TITLE
Hi，guanying. I have a question. I can't get results consistent with the paper based on your test code.

### ISSUE THREAD

**Issue by yangyuze1** [guanyingc/SDPS-Net#25]
------------ Options -------------
	benchmark: UPS_DiLiGenT_main
	bm_dir: data/datasets/DiLiGenT/pmsData
	bool_keys: ['int_aug', 'test_resc']
	color_aug: True
	color_ratio: 3
	concat_data: True
	crop: True
	crop_h: 128
	crop_w: 128
	cuda: True
	data_dir: I:/computervision/Photometric Stereo/PSMF-FCN-master/data/datasets/PS_Blobby_Dataset
	data_dir2: I:/computervision/Photometric Stereo/PSMF-FCN-master/data/datasets/PS_Sculpture_Dataset
	dataset: UPS_Synth_Dataset
	debug: False
	dir_int: False
	dirs_cls: 36
	epochs: 1
	fuse_type: max
	have_gt_n: False
	have_l_dirs: False
	have_l_ints: False
	in_img_num: 96
	in_light: False
	in_mask: True
	int_aug: True
	ints_cls: 20
	item: CVPR2019
	l_suffix: _mtrl.txt
	log: <utils.logger.Logger object at 0x000002329C254AC0>
	log_dir: data/models\4-18_run_model,LCNet_CVPR2019,LCNet,NENet,UPS_DiLiGenT_main,max,in_m-96,te_h-128,te_w-128,int_aug
	make_dir: True
	max_test_iter: -1
	max_train_iter: -1
	max_val_iter: -1
	model: LCNet
	model_s2: NENet
	multi_gpu: False
	noise: 0.05
	noise_aug: True
	normalize: False
	rand_sc: True
	rescale: True
	resume: None
	retrain: data/models/LCNet_CVPR2019.pth.tar
	retrain_s2: data/models/NENet_CVPR2019.pth.tar
	run_model: True
	s1_est_d: True
	s1_est_i: True
	s1_est_n: False
	s2_est_d: False
	s2_est_i: False
	s2_est_n: True
	s2_in_light: True
	save_intv: 1
	save_root: logdir/
	save_split: False
	scale_h: 128
	scale_w: 128
	seed: 0
	stage2: True
	str_keys: ['model', 'model_s2', 'benchmark', 'fuse_type']
	suffix: None
	test_batch: 1
	test_disp: 1
	test_h: 128
	test_resc: False
	test_save: 1
	test_save_n: 4
	test_w: 128
	time_sync: False
	train_disp: 20
	train_save: 200
	train_save_n: 4
	use_BN: False
	val_disp: 1
	val_intv: 1
	val_keys: ['in_img_num', 'test_h', 'test_w']
	val_save: 1
	workers: 0
-------------- End ----------------

=> fetching img pairs in 'data/UPS_DiLiGenT_main'
[test Data] 	10 objs 96 lights. Root: data/datasets/DiLiGenT/pmsData
Found Benchmark Data: 10 samples
	 Test Batch 1
[Network Input] Color image as input
[Network Input] Adding Mask as input
[Network Input] Input channel: 4
=> using pre-trained model 'data/models/LCNet_CVPR2019.pth.tar'
=> Model Parameters: 4409692
=> using pre-trained model_s2 'data/models/NENet_CVPR2019.pth.tar'
=> Stage2 Model Parameters: 2210048
---- Start test Epoch 1: 10 batches ----
 | TEST Iter [1/10] Epoch [1/1]
	 [Time 6.195s] ToCPU: 2.985s| ToGPU: 0.038s| Forward: 1.315s| 
	 [ACC] n_acc_11: 0.803| n_acc_30: 1.000| n_acc_45: 1.000| 
	 [ERR] l_err_mean: 7.317| n_err_mean: 7.012| 
	 [RATIO] ints_ratio: 0.055| 

 | TEST Iter [2/10] Epoch [1/1]
	 [Time 5.814s] ToCPU: 3.661s| ToGPU: 0.028s| Forward: 0.378s| 
	 [ACC] n_acc_11: 0.765| n_acc_30: 0.998| n_acc_45: 1.000| 
	 [ERR] l_err_mean: 5.449| n_err_mean: 7.594| 
	 [RATIO] ints_ratio: 0.053| 

 | TEST Iter [3/10] Epoch [1/1]
	 [Time 5.767s] ToCPU: 3.663s| ToGPU: 0.029s| Forward: 0.354s| 
	 [ACC] n_acc_11: 0.541| n_acc_30: 0.942| n_acc_45: 0.987| 
	 [ERR] l_err_mean: 10.870| n_err_mean: 12.987| 
	 [RATIO] ints_ratio: 0.066| 

 | TEST Iter [4/10] Epoch [1/1]
	 [Time 5.539s] ToCPU: 3.492s| ToGPU: 0.028s| Forward: 0.306s| 
	 [ACC] n_acc_11: 0.682| n_acc_30: 0.990| n_acc_45: 0.999| 
	 [ERR] l_err_mean: 6.336| n_err_mean: 10.041| 
	 [RATIO] ints_ratio: 0.095| 

 | TEST Iter [5/10] Epoch [1/1]
	 [Time 5.588s] ToCPU: 3.603s| ToGPU: 0.029s| Forward: 0.283s| 
	 [ACC] n_acc_11: 0.606| n_acc_30: 0.989| n_acc_45: 0.999| 
	 [ERR] l_err_mean: 12.351| n_err_mean: 10.644| 
	 [RATIO] ints_ratio: 0.099| 

 | TEST Iter [6/10] Epoch [1/1]
	 [Time 5.584s] ToCPU: 3.512s| ToGPU: 0.029s| Forward: 0.368s| 
	 [ACC] n_acc_11: 0.578| n_acc_30: 0.969| n_acc_45: 0.991| 
	 [ERR] l_err_mean: 11.416| n_err_mean: 11.780| 
	 [RATIO] ints_ratio: 0.092| 

 | TEST Iter [7/10] Epoch [1/1]
	 [Time 5.478s] ToCPU: 3.450s| ToGPU: 0.029s| Forward: 0.316s| 
	 [ACC] n_acc_11: 0.332| n_acc_30: 0.807| n_acc_45: 0.906| 
	 [ERR] l_err_mean: 16.516| n_err_mean: 20.532| 
	 [RATIO] ints_ratio: 0.115| 

 | TEST Iter [8/10] Epoch [1/1]
	 [Time 5.690s] ToCPU: 3.653s| ToGPU: 0.028s| Forward: 0.289s| 
	 [ACC] n_acc_11: 0.796| n_acc_30: 0.980| n_acc_45: 0.995| 
	 [ERR] l_err_mean: 5.316| n_err_mean: 8.478| 
	 [RATIO] ints_ratio: 0.073| 

 | TEST Iter [9/10] Epoch [1/1]
	 [Time 5.613s] ToCPU: 3.583s| ToGPU: 0.028s| Forward: 0.299s| 
	 [ACC] n_acc_11: 0.792| n_acc_30: 0.982| n_acc_45: 0.995| 
	 [ERR] l_err_mean: 7.804| n_err_mean: 8.431| 
	 [RATIO] ints_ratio: 0.102| 

 | TEST Iter [10/10] Epoch [1/1]
	 [Time 5.588s] ToCPU: 3.576s| ToGPU: 0.028s| Forward: 0.312s| 
	 [ACC] n_acc_11: 0.067| n_acc_30: 0.510| n_acc_45: 0.719| 
	 [ERR] l_err_mean: 19.178| n_err_mean: 36.334| 
	 [RATIO] ints_ratio: 0.267| 

---------- TEST Epoch 1 Summary -----------
	 [ACC] n_acc_11: 0.596| n_acc_30: 0.917| n_acc_45: 0.959| 
	 [ERR] l_err_mean: 10.255| n_err_mean: 13.383| 
	 [RATIO] ints_ratio: 0.102|

**Comment by guanyingc** [comment]
Hi, could you show me the command you ran and some sample results?

**Comment by yangyuze1** [comment]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "guanyingc/SDPS-Net"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月18日(星期二) 晚上7:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [guanyingc/SDPS-Net] Hi，guanying. I have a question. I can't get results consistent with the paper based on your test code. (Issue #25)





 
Hi, could you show me the command you ran and some sample results?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;


从QQ邮箱发来的超大附件

4-18_run_model,LCNet_CVPR2019,LCNet,NENet,UPS_DiLiGenT_main,max,in_m-96,te_h-128,te_w-128,int_aug.rar (3.77M, 无限期)进入下载页面：https://mail.qq.com/cgi-bin/ftnExs_download?k=ac396261589af19d84d61e231638004d4e4d0356535c54561b585b540915575a020d4f03020a014f53585458510d0b5055585a54305d32561b085a3e424d5c3d5b5606045c147e21785c163e736e6230040953581c74712c534d4e2f757657161a6c32326f7c5b2e5f7e070f64675f035f574e0c51401e0b58660f4c090e1e1653660a4c010a0a4e425c3d161d09005a1a500c156f594705184b03133005&t=exs_ftn_download&code=69ba082b

**Comment by yangyuze1** [comment]
你好，冠英。我在linux系统和windows系统上都运行过，结果是一致的，都未达到您论文的精度。我不知道哪里出错了，代码也没有报错。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "guanyingc/SDPS-Net"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月18日(星期二) 晚上7:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [guanyingc/SDPS-Net] Hi，guanying. I have a question. I can't get results consistent with the paper based on your test code. (Issue #25)





 
Hi, could you show me the command you ran and some sample results?

**Comment by yangyuze1** [comment]
I found the problem, the error of testing with the data in the original 'pmsData' would be very large, testing with the cropped 'pmsData_crop' data, the error can reach the experimental results of the paper. The background has a very large impact on the network.

**Comment by guanyingc** [comment]
Sorry, I missed this issue as I am very busy recently. Glad to hear that! Yes, you should first crop the image for testing as the preprocessing code.
